e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -507,6 +451,102 @@ describe("scenarios > dashboards > filters > auto apply", () => {
         undoToast().should("not.exist");
       });
     });
+
+    describe("full-app embeddings", () => {
+      beforeEach(() => {
+        cy.signInAsNormalUser();
+      });
+
+      it("should apply filters after clicking the apply button when auto-apply filters is turned off", () => {
+        createDashboard({
+          dashboardDetails: {
+            name: "Full-app embedding dashboard",
+            auto_apply_filters: false,
+          },
+        });
+        cy.get("@dashboardId").then(dashboardId => {
+          visitFullAppEmbeddingUrl({
+            url: `/dashboard/${dashboardId}`,
+            qs: { side_nav: false, logo: false },
+          });
+        });
+        cy.findByDisplayValue("Full-app embedding dashboard").should(
+          "be.visible",
+        );
+        // Ensure that we're viewing the dashboard in full-app embedding mode, since `logo` is a full-app embedding parameter.
+        cy.findByTestId("main-logo").should("not.exist");
+
+        dashboardParametersContainer().button("Apply").should("not.exist");
+        filterWidget().findByText("Category").click();
+        popover().within(() => {
+          cy.findByText("Widget").click();
+          cy.button("Add filter").click();
+        });
+        getDashboardCard().findByText("Rows 1-5 of 200").should("be.visible");
+        dashboardParametersContainer()
+          .button("Apply")
+          .should("be.visible")
+          .click();
+        getDashboardCard().findByText("Rows 1-5 of 54").should("be.visible");
+      });
+
+      it("should display a toast when a dashboard takes longer than 15s to load", () => {
+        createDashboard();
+        // Not sure why I need to pass a date in this case, but it doesn't work without it.
+        cy.clock(Date.now());
+        openSlowFullAppEmbeddingDashboard({ [FILTER.slug]: "Gadget" });
+        cy.tick(TOAST_TIMEOUT);
+        cy.wait("@cardQuery");
+        undoToast().within(() => {
+          cy.findByText(TOAST_MESSAGE).should("be.visible");
+          cy.button("Turn off").click();
+          cy.wait("@updateDashboard");
+        });
+
+        // In embedding we'll load bookmark after the dashboard is loaded, it's the opposite in normal app because bookmark is cached from somewhere else.
+        // And somehow, dashboard card query will be completed before the dashboard even start to load, and in entity loader it uses `setTimeout`,
+        // so to make sure callback in `setTimeout` is called, we need to advance the clock using cy.tick().
+        cy.tick();
```

Comment:
You can try to remove this line, and the dashboard header will never finish loading. The reason is that the API somehow starts kicking in after we call the last `cy.tick`, so we need to advance the clock here as well since entity loader uses `setTimeout` under the hood.

https://github.com/metabase/metabase/blob/fd86ce3fb50b43340e4ac9f46d473a86b01350c0/frontend/src/metabase/lib/redux/utils.js#L235-L238

Without advancing the clock, the action to set the request status to loaded will never be called.